src/main/java/org/codehaus/mojo/build/CreateTimestampMojo.java

```diff
@@ -106,12 +112,38 @@ public void execute()
 
         timestampString = Utils.createTimestamp( timestampFormat, timezone );
 
-        getLog().debug( "Storing timestamp property: " + timestampPropertyName + " " + timestampString );
-
-        for ( MavenProject project : session.getProjectDependencyGraph().getSortedProjects() )
+        if (getLock())
```

Comment:
if not getting lock we do not write the timestamp anymore?

Reply:
How can this happen?

Reply:
looking at the code `getLock() ` can return `false` :)

Reply:
Yes, but I think one thread will always get a lock, which is sufficient. That's why I was asking.
Can you picture a scenario where NO thread could get a lock?

Reply:
@olamy ping